app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -16,60 +21,159 @@
         ButterKnife.bind(this, getActivity());
-        vLoginButton.setEnabled(false);
-    }
+        Context context = getActivity();
+
+        sharedPref = context.getSharedPreferences(
+                getString(R.string.preferences_name), Context.MODE_PRIVATE);
+
+        editor = sharedPref.edit();
+
+        vTermsConditionsText.setMovementMethod(LinkMovementMethod.getInstance());
+
+        vEmailInput.setText(sharedPref.getString(getString(R.string.login_email), ""));
+        vPasswordInput.setText(sharedPref.getString(getString(R.string.login_pass), ""));
+    } // public void init()
 
     /**
      *
      */
     public void setListeners() {
-        vFirstNameInput.addTextChangedListener(new TextWatcher() {
+        vEmailInput.addTextChangedListener(new TextWatcher() {
             @OnTextChanged
             public void beforeTextChanged(CharSequence s, int start, int count, int after) {
             }
 
             @OnTextChanged
             public void onTextChanged(CharSequence s, int start, int before, int count) {
-                if (!vFirstNameInput.getText().toString().equals("")) {
-                    vLoginButton.setEnabled(true);
-                } else {
-                    vLoginButton.setEnabled(false);
-                }
+                vEmailInput.setError(null);
             }
 
             @OnTextChanged
             public void afterTextChanged(Editable s) {
             }
         });
 
+        vEmailInput.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
+                    if (!validateEmail(vEmailInput.getText().toString())) {
+                        vEmailInput.setError(getText(R.string.login_email_invalid));
+                    }
+                }
+            }
+        });
+
+        vPasswordInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                vEmailInput.setError(null);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+
         vLoginButton.setOnClickListener(new View.OnClickListener() {
+
             @OnClick
             public void onClick(View v) {
-                presenter.storeUsername(vFirstNameInput.getText().toString());
-            }
-        });
-    }
+                if (validateFields()) {
+                    editor.putString(getString(R.string.login_email), vEmailInput.getText().toString());
+                    editor.putString(getString(R.string.login_pass), vPasswordInput.getText().toString());
+                    editor.commit();
+
+                    presenter.onUsernameSaved();
```

Comment:
Same as above. Fragment it's a silly class, so we can move everything to presenter and here do something like 
`presenter.onLoginButtonClicked()`